File: p9/p9.erl
#!/usr/bin/env escript
%% -*- erlang -*-
%%! -smp enable -sname p9
% vim:syn=erlang

-mode(compile).

main(_) ->
    %D = 1000,
    %Answer = lists:last([ X*Y*(D-X-Y) || X <- lists:seq(1, D), Y <- lists:seq(1, D), X*X + Y*Y =:= (D-X-Y)*(D-X-Y)]),
    Answer = a(1),
    io:format("Answer: ~p ~n", [Answer]).

a(X) when X < 1000 ->
    case b(X,X) of
        false -> a(X+1);
        Ret   -> Ret
    end.

b(X,Y) when Y < 1000, X*X+Y*Y =:= (1000-X-Y)*(1000-X-Y) -> X*Y*(1000-X-Y);
b(X,Y) when Y < 1000 -> b(X,Y+1);
b(_,_) -> false.

File: p9/p9.pl
#!/usr/bin/perl -w
use strict;

for my $a (1..1000){
    for my $b ($a..1000){
        my $c = 1000 - $a - $b;
        if($c > 0 && ($a*$a+$b*$b == $c*$c)){
            print "Answer: ".$a*$b*$c."\n";
            last;
        }
    }    
}
